File: LeetCode/409. Longest Palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        HashSet<Character> set = new HashSet<>();
        int res = 0;
        for (char c : s.toCharArray()) {
            if (set.contains(c)) {
                set.remove(c);
                res += 2;
            } else set.add(c);
        }
        if (!set.isEmpty()) res++;
        return res;
    }
} //TC: O(n), SC: O(n), where n is the length of s

File: LeetCode/409. Longest Palindrome.kt
class Solution {
    fun longestPalindrome(s: String): Int {
        var res = 0
        val set = HashSet<Char>()

        s.forEach {
            if (it in set) { //if the char is in the set
                set.remove(it)
                res+=2 //remove it and add result by 2
            } else { //if not add it to the set
                set.add(it)
            }
        }

        return if (set.isNotEmpty()) ++res //if the set is not empty, it means there is one or more chars not in pair, add result by 1 as a char in the middle of a palindrome
               else res //otherwise just return the result
    }
}
